crates/solver/src/settlement.rs

```diff
@@ -453,6 +453,17 @@ impl Settlement {
     pub fn encode(self, internalization_strategy: InternalizationStrategy) -> EncodedSettlement {
         self.encoder.finish(internalization_strategy)
     }
+
+    pub fn encode_uninternalized_if_different(self) -> Option<EncodedSettlement> {
+        if self.encoder.contains_internalized_interactions() {
+            Some(
+                self.encoder
+                    .finish(InternalizationStrategy::EncodeAllInteractions),
+            )
+        } else {
+            None
+        }
+    }
```

Comment:
```suggestion
        self.encoder.contains_internalized_interactions().then_some(
            self.encoder
                .finish(InternalizationStrategy::EncodeAllInteractions),
        )
```

nit: Found a cool one-liner but feel free to ignore since I'm not sure if it improves the readibility.

Reply:
😍 - that is awesome and didn't know it was a thing!